www/assets/js/sofia.data.js

```diff
@@ -341,6 +345,41 @@ S.data = {
                     //TODO handle errors
                 });
                 */
+              },
+              showConfigurationModal: function () {
+                  S.tool.getDialog("#show-config-dialog").showModal();
+              },
+              closeConfigurationModal: function () {
+                  this.db = S.config.db; //Reset
+                  S.tool.getDialog("#show-config-dialog").close();
+              },
+              scanQRCode: function (event) {
+                var el = this;
+                cordova.plugins.barcodeScanner.scan(
+                    function (result) {
+                      console.log("We got a barcode\n" +
+                            "Result: " + result.text + "\n" +
+                            "Format: " + result.format + "\n" +
+                            "Cancelled: " + result.cancelled);
+                      if(!result.cancelled){
+                        if(result.format != "QR_CODE"){
+                          alert("Format "+result.format+" incorrect !")
+                        }else{
+                          var tmp = result.text.split("/");
+                          el.db.name = tmp.pop()
+                          el.db.url = tmp.join('/')
+                          //TODO support username and password
+                        }
+                      }
+                    },
+                    function (error) {
+                        alert("Scanning failed: " + error);
+                    }
+                 );
+              },
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ["event" is defined but never used](https://www.codacy.com/app/sapk/SOFIA/file/2059618678/issues/source?bid=2990559&fileBranchId=3135266#l380)